<!doctype html>
<!-- paulirish.com/2008/conditional-stylesheets-vs-css-hacks-answer-neither/ -->
<!--[if lt IE 7]> <html class="no-js ie6 oldie" lang="en"> <![endif]-->
<!--[if IE 7]>    <html class="no-js ie7 oldie" lang="en"> <![endif]-->
<!--[if IE 8]>    <html class="no-js ie8 oldie" lang="en"> <![endif]-->
<!-- Consider adding an manifest.appcache: h5bp.com/d/Offline -->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en"> <!--<![endif]-->
<head>
	<meta charset="utf-8">

	<!-- Use the .htaccess and remove these lines to avoid edge case issues.
		More info: h5bp.com/b/378 -->
	<meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1">

	<title>BCB 504, Applied Bioinformatics</title>
	<meta name="description" content="Matt Settles, Bioinformatician, Genomics Resources Core, IBEST, University of Idaho, Applied bioinformatics">
	<meta name="author" content="Matt Settles">

	<!-- Mobile viewport optimized: j.mp/bplateviewport -->
	<meta name="viewport" content="width=device-width,initial-scale=1">

	<!-- Place favicon.ico and apple-touch-icon.png in the root directory: mathiasbynens.be/notes/touch-icons -->

	<!-- CSS: implied media=all -->
	<!-- CSS concatenated and minified via ant build script-->
	<link rel="stylesheet" href="../../css/style.css">
	<!-- end CSS-->

	<!-- More ideas for your <head> here: h5bp.com/d/head-Tips -->

	<!-- All JavaScript at the bottom, except for Modernizr / Respond.
		Modernizr enables HTML5 elements & feature detects; Respond is a polyfill for min/max-width CSS3 Media Queries
		For optimal performance, use a custom Modernizr build: www.modernizr.com/download/ -->
	<script src="../../js/libs/modernizr-2.0.6.min.js"></script>
</head>

<body>

	<section id="page">
		<!-- Header Content -->
		<header id="banner">
			<h1 style="font-size:3.0em">BCB 504, Applied Bioinformatics</h1>
			<h2>University of Idaho, Spring 2012</h2>
			<h2>Instructor: Matt Settles</h2>
			<h2>Teaching Assistant: Matt Pennell</h2>
			<nav> <!-- The nav link semantically marks your main site navigation -->
				<ul>
					<li><a href="#description">Description</a></li>
					<li><a href="#calendar">Calendar</a></li>
					<li><a href="#reading">Reading</a></li>
					<li><a href="#datasets">Datasets</a></li>
					<li><a href="#projects">Projects</a></li>
					<li><a href="#documents">Documents</a></li>
					<li><a href="#links">Links</a></li>
				</ul>
			</nav>
			<div class="clearfix"></div>
			<div class="line"></div>  <!-- Dividing line -->
		</header>

		<!-- Page Body -->
		<!-- Row 1 -->
		<div id="row" role="row">
			<!-- Description \ start -->
			<article id="description" >
				<h2>Course Description and Syllabus</h2>
				<div class="line"></div>
				<div class="articleBody">
					<p><strong>Catalog Description:</strong> A data driven approach for the computational and statistical understanding and expertise needed to solve bioinformatics problems that you will likely encounter in your research. Topics will include: microarray data analysis, high-throughput sequence data analysis and SNP genotyping analysis as well as some additional specific advanced topics.</p>
					<p><strong>Class Meetings:</strong> Monday/Wednesday 10:00-11:30 in LSS 440 (IBEST Classroom)</p>
					<p><strong>Course Credits:</strong> 3 cr.</p>
					<p><strong>Prerequisites:</strong> One of CS 120 (Computer Science I), Stat 452 (Mathematical Statistics), Biology 456 (Computer Skills for Biologists), or with permission.</p>
					<p><strong>Textbook:</strong> None</p>
					<p><strong>Instructor:</strong> Matt Settles</p>
					<p><strong>Office:</strong> LSS 441C; <strong>Phone:</strong>885-6051; <strong>Email:</strong><a href="mailto:msettles@uidaho.edu">msettles@uidaho.edu</a></p>

					<p><strong>Teaching Assistant:</strong> Matt Pennell</p>
					<p><strong>Office:</strong>CNR 212;<strong>Phone:</strong>208-874-7539; <strong>Email:</strong><a href="mailto:mwpennell@gmail.com">mwpennell@gmail.com</a></p>

					<p><strong>Course Goals:</strong> Following this course the student will be capable of performing their own data analysis project, understanding the technical and statistical tools needed to conduct the analysis with the computational ability to do so, and critically review and implement techniques and methods in publications.</p>
					<p><strong>Course Format:</strong> The course will be divided into both lecture and lab/workshop sessions.</p>

					<p>Topics will include:</p>
					<ul>
						<li>Expression microarray analysis</li>
						<li>CGH/CHiP-Chip microarray analysis</li>
						<li>Phylogenetic methods</li>
						<li>High-throughput sequence assembly</li>
						<li>High-throughput sequence mapping</li>
						<li>RNA-seq studies</li>
						<li>Metagenomics</li>
						<li>Whole Genome Association Studies</li>
					</ul>
					<p><strong>Course Grading:</strong>A point system will be used for grading. Your semester grade will be based on a standard grading curve (90%, 80%, 70%,â€¦) of the cumulative number of points you have earned by the last day of finals week. There will be 6 projects (each worth a 12 points), and roughly 14 publication reviews (each worth 2 points), for a total of 100 points.</p>

					<p>Each project will be a report on the analysis of public data (or your own data) using the techniques discussed in class. The reports must be written using Latex with embedded R code of the complete analysis. A template with brief introduction is provided in the documents section below.</p>

					<p>Publication reviews will be short 1/2 to full page comments on assigned methods papers. A Template is provided in the documents section below.</p>
				</div>
			</article>
			<!-- Description end -->
			<div class="clearfix"></div>
		</div>

		<!-- Row 2 -->
		<div id="row" role="row">
			<!-- Calendar start -->
			<article id="calendar"> 
				<h2>Calendar</h2>
				<div class="line"></div>
				<div class="articleBody">
					<center><iframe src="https://www.google.com/calendar/embed?showCalendars=0&amp;mode=AGENDA&amp;height=600&amp;wkst=1&amp;bgcolor=%23ffffff&amp;src=llvkcnih5qbgql8son718k556c%40group.calendar.google.com&amp;color=%232F6309&amp;ctz=America%2FLos_Angeles" style=" border-width:0 " width="900" height="600" frameborder="0" scrolling="no"></iframe></center>
				    <p>Links: <a href="https://www.google.com/calendar/ical/llvkcnih5qbgql8son718k556c%40group.calendar.google.com/public/basic.ics">ICAL</a>, <a href="https://www.google.com/calendar/feeds/llvkcnih5qbgql8son718k556c%40group.calendar.google.com/public/basic">XML</a>, <a href="https://www.google.com/calendar/embed?src=llvkcnih5qbgql8son718k556c%40group.calendar.google.com&ctz=America/Vancouver">HTML</a></p>
				</div>
			</article>
			<!-- Calendar End -->
			<div class="clearfix"></div>
		</div>

		<!-- Row 3 -->
		<div id="row" role="row">
			<!-- Reading start -->
			<article id="reading">
				<h2>Assigned Reading</h2>
				<div class="line"></div>
				<div class="articleBody">
					<p>Rob Jelier, Jelle Geoman, Kristina Hettne, Martin Schuemie, Johan Dunnen and Peter Hoen. <i>Literature-aided interpretation of gene expression data with the weighted global test.</i> Briefing in Bioinformatics, 12(5):518-529, 2010. (<a href="reading/Brief Bioinform-2011-Jelier-518-29.pdf">pdf</a>)</p>
				    <p>Yao Yu, Kang Tu, Siyuan Zheng, Yun Li, Guohui Ding, Jie Ping, Pei Hao and Yixue Li. <i>GEOGLE: context mining tool for the correlation between gene expression and the phenotypic distinction.</i> BMC Bioinformatics, 10:264, 2009. (<a href="reading/1471-2105-10-264.pdf">pdf</a>)</p>
				    <p>Rafael Irizarry, Christine Ladd-Acosta, Benilton Carvalho, Hao Wu, Sheri Brandenburg, Jeffrey Jeddeloh, Bo Wen, and Andrew Feinberg. <i> Comprehensive high-throughput arrays for relative methylation (CHARM)</i> Genome Research, 18: 780-790, 2008. (<a href="reading/Genome Res.-2008-Irizarry-780-90.pdf">pdf</a>)</p>
				</div>
			</article>
			<!-- Reading end -->
			<div class="clearfix"></div>
		</div>

		<!-- Row 3 -->
		<div id="row" role="row">
			<!-- Data start -->
			<article id="datasets">
				<h2>Example Datasets</h2>
				<div class="line"></div>
				<div class="articleBody">
					<p><a href="http://webpages.uidaho.edu/msettles/largefiles/Selenium.zip">Affymetrix microarray (Robison/Benner Selenium Exp)</a></p>
					<p><a href="http://webpages.uidaho.edu/msettles/largefiles/Microarray_Nim_1Sept2011.zip">Nimblegen microarray (Hartzel Exp)</a></p>
				</div>
			</article>
			<!-- Data end -->
			<div class="clearfix"></div>
		</div>

		<!-- Row 4 -->
		<div id="row" role="row">
			<!-- Project start -->
			<article id="projects">
				<h2>Assigned Projects</h2>
				<div class="line"></div>
				<div class="articleBody">
					<p><a href="codeprojects/affymetrix_part1.R">Affymetrix R code (QA/Background correction/filter/diff exp)</a></p>
					<p><a href="codeprojects/nimblegen_part1.R">Nimblegen R code (QA/Background correction/filter/diff exp)</a></p>
				</div>
			</article>
			<!-- Projects end -->
			<div class="clearfix"></div>
		</div>

		<!-- Row 5 -->
		<div id="row" role="row">
			<!-- Documents start -->
			<article id="documents", class="article-left"> 
				<h2>Documents</h2>
				<div class="line"></div>
				<div class="articleBody">
					<p><a href="sweaveTemplate/sweaveTemplate.tex">Sweave Template</a></p>
					<p><a href="https://docs.google.com/open?id=0B_3lTFvg_WhxZjVjN2Y5MTgtMGRhZi00MWZjLThlNzktMzQ1NzU3YjAwMDM2">Publication review template</a></p>
				</div>
			</article>
			<!-- Documents end -->

			<!-- Links start -->
			<article id="links", class="article-right">
				<h2>Links</h2>
				<div class="line"></div>
				<div class="articleBody">
					<p><a href="http://www.stat.uni-muenchen.de/~leisch/Sweave/">The Sweave Homepage</a></p>
					<p><a href="http://tobi.oetiker.ch/lshort/lshort.pdf">The not so short introduction to LaTeX</a></p>
					<p><a href="http://www.r-project.org/">R - Homepage</a></p>
					<p><a href="http://www.bioconductor.org">Bioconductor - Homepage</a></p>
					<p><a href="http://rstudio.org/">R Studio</a></p>
					<p><a href="http://cran.r-project.org/other-docs.html">R contributed documentation</a></p>
					<p><a href="http://files.fosswire.com/2007/08/fwunixref.pdf">Linux cheat sheet</a></p>
				</div>
			</article>
			<!-- Links end -->
			<div class="clearfix"></div>
		</div>

		<!-- Footer Section -->
		<footer>
			<nav> <!-- The nav link semantically marks your main site navigation -->
				<ul>
					<li><a href="#description">Description</a></li>
					<li><a href="#calendar">Calendar</a></li>
					<li><a href="#reading">Reading</a></li>
					<li><a href="#datasets">Datasets</a></li>
					<li><a href="#projects">Projects</a></li>
					<li><a href="#documents">Documents</a></li>
					<li><a href="#links">Links</a></li>
				</ul>
			</nav>
			<div class="clearfix"></div>
			<div class="line"></div>
			<p>Copyright 2012 - Matt Settles</p> <!-- Change the copyright notice -->
			<a href="#" class="up">Go UP</a>
		</footer>
	</section> <!-- end of #page -->

	<!-- JavaScript at the bottom for fast page loading -->

	<!-- Grab Google CDN's jQuery, with a protocol relative URL; fall back to local if offline -->
	<script src="//ajax.googleapis.com/ajax/libs/jquery/1.6.2/jquery.min.js"></script>
	<script>window.jQuery || document.write('<script src="../../js/libs/jquery-1.6.2.min.js"><\/script>')</script>
	<!-- Smooth Scrolling with jquery.scollTo-min.js -->
	<script src="../../js/mylibs/jquery.scrollTo-1.4.2/jquery.scrollTo-min.js"></script>

	<!-- scripts concatenated and minified via ant build script-->
	<script defer src="../../js/plugins.js"></script>
	<script defer src="../../js/script.js"></script>
	<!-- end scripts-->

  <!-- Change UA-XXXXX-X to be your site's ID -->
	<script>
		window._gaq = [['_setAccount','UA-28001006-1'],['_trackPageview'],['_trackPageLoadTime']];
		Modernizr.load({
			load: ('https:' == location.protocol ? '//ssl' : '//www') + '.google-analytics.com/ga.js'
		});
	</script>

	<!-- Prompt IE 6 users to install Chrome Frame. Remove this if you want to support IE 6.
	chromium.org/developers/how-tos/chrome-frame-getting-started -->
	<!--[if lt IE 7 ]>
	<script src="//ajax.googleapis.com/ajax/libs/chrome-frame/1.0.3/CFInstall.min.js"></script>
	<script>window.attachEvent('onload',function(){CFInstall.check({mode:'overlay'})})</script>
	<![endif]-->
</body>
</html>