<!doctype html>
<!-- paulirish.com/2008/conditional-stylesheets-vs-css-hacks-answer-neither/ -->
<!--[if lt IE 7]> <html class="no-js ie6 oldie" lang="en"> <![endif]-->
<!--[if IE 7]>    <html class="no-js ie7 oldie" lang="en"> <![endif]-->
<!--[if IE 8]>    <html class="no-js ie8 oldie" lang="en"> <![endif]-->
<!-- Consider adding an manifest.appcache: h5bp.com/d/Offline -->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en"> <!--<![endif]-->
<head>
	<meta charset="utf-8">

	<!-- Use the .htaccess and remove these lines to avoid edge case issues.
	More info: h5bp.com/b/378 -->
	<meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1">

	<title>Matt Settles, Bioinformatician</title>
	<meta name="description" content="Matt Settles, Bioinformatician, Genomics Resources Core, IBEST, University of Idaho, personnal web page">
	<meta name="author" content="Matt Settles">

	<!-- Mobile viewport optimized: j.mp/bplateviewport -->
	<meta name="viewport" content="width=device-width,initial-scale=1">

	<!-- Place favicon.ico and apple-touch-icon.png in the root directory: mathiasbynens.be/notes/touch-icons -->

	<!-- CSS: implied media=all -->
	<!-- CSS concatenated and minified via ant build script-->
	<link rel="stylesheet" href="css/style.css">
	<!-- end CSS-->

	<!-- More ideas for your <head> here: h5bp.com/d/head-Tips -->

	<!-- All JavaScript at the bottom, except for Modernizr / Respond.
		Modernizr enables HTML5 elements & feature detects; Respond is a polyfill for min/max-width CSS3 Media Queries
		For optimal performance, use a custom Modernizr build: www.modernizr.com/download/ -->
	<script src="js/libs/modernizr-2.0.6.min.js"></script>
	<!-- Google Maps API -->
	<script type="text/javascript" src="http://maps.google.com/maps/api/js?sensor=false"></script>
	<script type="text/javascript">
		(function() {
			window.onload = function(){
				var latlng = new google.maps.LatLng(46.727213,-117.010325);
				var options = {
					zoom: 17,
					center: latlng,
					mapTypeId: google.maps.MapTypeId.SATELLITE
				}; 
				var map = new google.maps.Map(document.getElementById('map'), options);
				var marker = new google.maps.Marker({
					position: latlng,
					title:"Life Sciences South!"
				});
				marker.setMap(map);
			}
		})();
	</script>

</head>

<body>

	<section id="page">
		<!-- Header Content -->
		<header id="banner">
			<h1>Matt Settles 
			<h2>Research Assistant Professor, Depeartment of Biological Sciences</h2>
			<h2>Director, IBEST Genomics Resources Core Facility</h2>
			<h2>University of Idaho, Moscow, Idaho</h2>
			<nav> <!-- The nav link semantically marks your main site navigation -->
				<ul>
					<li><a href="#contact">Contact</a></li>
					<li><a href="#research">Research/Interests</a></li>
					<li><a href="#courses">Courses</a></li>
					<li><a href="#software">Software</a></li>
					<li><a href="#links">Links</a></li>
					<li><a href="#publications">Publications</a></li>
				</ul>
			</nav>
			<div class="clearfix"></div>
			<div class="line"></div>  <!-- Dividing line -->
		</header>

		<!-- Page Body -->
		<!-- Row 1 -->
		<div id="row" role="row">
			<!-- Contact Information start -->
			<article id="contact" >
				<h2>Contact and Information</h2>
				<div class="line"></div>
				<div class="articleBody">
					<div id="map"></div>
					<div class="contact">
						<p><strong>Email:</strong> <a href="mailto:msettles@uidaho.edu">msettles@uidaho.edu</a></p>
						<p><strong>Phone:</strong> 208-885-6051</p>
						<p><strong>Fax:</strong> 208-885-5003</p>
						<p style="margin-top:1.0em"><strong>Office:</strong></p>
						<p style="text-indent:1em"><a href="http://www.uidaho.edu/research/ibest">Institute for Bioinformatics and Evolutionary Studies</a></p>
						<p style="text-indent:1em"><a href="http://www.uidaho.edu/sci/biology">Department of Biological Sciences</a></p>
						<p style="text-indent:1em"><a href="http://www.uidaho.edu/">University Of Idaho</a></p>
						<p style="text-indent:1em">Life Science South | Room 441C</p>
						<p style="text-indent:1em">PO Box 443051</p>
						<p style="text-indent:1em">Moscow, Idaho 83843-3051</p>
						<p style="margin-top:1.0em"><strong>Education:</strong></p>
						<p style="text-indent:1em">Ph.D. Bioinformatics and Computational Biology, University of Idaho, 2011</p>
						<p style="text-indent:1em">M.S. Computer Science, University of Idaho, 2005</p>
						<p style="text-indent:1em">B.S. Electrical Engineering, University of Portland, 2002</p>
					</div>
				</div>
			</article>
		    <!-- Contact Information end -->
			<div class="clearfix"></div>
		</div>

		<!-- Row 2 -->
		<div id="row" role="row">
			<!-- Research/Interests start -->
			<article id="research"> 
				<h2>Research and Interests</h2>
				<div class="line"></div>
				<div class="articleBody">
					<h4>Applied Bioinformatics</h4>
					<p>Over the past decade and a half Iâ€™ve accumulated a significant amount of experience across a wide spectrum of research topics, both computational and biological, with the unifying theme of "big data". Today I consider myself a genomicist; meaning, I conduct research in the three primary facets of genome research: data generation, data analysis and new method development. My research path has progressed from computational and problem solving algorithms in my undergraduate and masters years to applied bioinformatic analysis and new method development during my PhD and research associate appointments and finally to where I am today, as an genomics experimentalist and Director of a state of the art genomics core facility. My current research is on the generation, computational manipulation and interpretation of very large data sets across a wide range of disciplines, often applying techniques not originally designed for a particular data type or experiment in order to ask, and answer, new and interesting biological questions. I have had the pleasure of collaborating on projects in a wide range of biological subjects and problems including: functional genomics, evolutionary genomics, behavior genomics, comparative genomics, human and environmental microbiome, epigenomics, whole genome association studies and many others.</p>
					<h4>Microarrays - Single Feature Polymorphisms</h4>
					<p>Gene expression microarrays are a tool used by ecologists, to study the genome-wide transcriptional variation that underlies complex interactions among and between organisms and their environment. Microarrays allow for the simultaneous measurement of thousands of gene products across many samples. These measurements rely on sequence similarity for efficient hybridization of sample mRNA targets to the microarray probes. Genetic polymorphisms located within a microarray's probe affects this hybridization efficiency. These differences in probe level hybridization efficiencies are known as single feature polymorphisms (SFPs). SFPs are both a source of error, they negatively impact the estimate for transcript abundance, and a source of information, they are genetic markers. The Affymetrix 3' IVT microarray platform uses 11 to 16 short-oligonucleotide probes, covering different regions of a gene, to measure transcript abundance. The redundancy of probes within the same gene, allows for the capability to detect candidate SFPs from the data itself. My previous work presented three new results pertaining to SFPs in Affymetrix gene expression microarray data. The first result is a novel, improved algorithm for the accurate detection of SFPs. The second result evaluates the impact of SFPs to false positives in both differential expression analysis and the detection of SFPs themselves and offers a solution to reduce false positives induced by SFPs. Finally, the third result demonstrates how SFPs can be used to provide a new means for evaluating the accuracy of critical microarray preprocessing steps. These results will provide ecologists with new tools and techniques when using Affymetrix 3' IVT microarrays in ecological microarray studies.</p>
					<h4>Breeding Swarms: A Genetic Algorithm/Particle Swarm Optimization Hybrid Algorihtm</h4>
					<p>Genetic Algorithms and Particles Swarm Optimization are both population based stochastic search algorithms that have proven to be successful in solving very difficult problems. However, both models have strengths and weaknesses. Previous researchers have suggested that a hybrid of the GA and PSO models could lead to further advances. Breeding Swarms, combines the strengths of GAs with those of PSO. The Breeding Swarms algorithm improves search results for real valued problems by controlling the amount of diversity in the population. We performed experiments on optimizing standard benchmark functions and recurrent neural networks. Results show that the Breeding Swarm algorithm produces significantly better results than either GA or PSO, scales well to large problems and is more consistent over different problem domains.</p>
				</div>
			</article>
			<!-- Research/Interests End -->
			<div class="clearfix"></div>
		</div>

		<!-- Row 3 -->
		<div id="row" role="row">
			<!-- Courses start -->
			<article id="courses", class="article-left"> 
				<h2>Courses</h2>
				<div class="line"></div>
				<div class="articleBody">
					<p><a href="courses/bcb511Sp15/index.html">BCB 511: Applied Bioinformatics -- Spring 2015</a></p>
					<p><a href="courses/bcb504Sp13/index.html">BCB 504: Applied Bioinformatics -- Spring 2013</a></p>
					<p><a href="courses/bcb504Sp12/index.html">BCB 504: Applied Bioinformatics -- Spring 2012</a></p>
				</div>
			</article>
			<!-- Courses end -->

			<!-- Software start -->
			<article id="software", class="article-right">
				<h2>Software</h2>
				<div class="line"></div>
				<div class="articleBody">
					<p><a href="https://github.com/msettles/dbcAmplicons">dbcAmplicons</a>: Analsyis of double-barcoded massively multiplexed Illumina amplicons</p>
					<p><a href="https://github.com/ibest/ARC">ARC</a>: ARC is a pipeline which facilitates iterative, reference guided de novo assemblies.</p>
					<p><a href="https://bitbucket.org/izhbannikov/seqyclean">SeqyClean</a>: a Software Tool for Comprehensive Preprocessing of Sequence Data.</p>
					<p><a href="http://www.bioconductor.org/packages/release/bioc/html/OTUbase.html">OTUbase</a>: Provides a platform for Operational Taxonomic Unit based analysis within R.</p>
					<p><a href="https://github.com/msettles/rSFFreader">rSFFreader</a>: rSFFreader is a R package to read in sequence data (sequence, quality and clip points) from both Roche 454 sequencers and Life Sciences Ion Torrent Seuqencers.</p>
				</div>
			</article>
			<!-- Software end -->
			<div class="clearfix"></div>
		</div>

		<!-- Row 4 -->
		<div id="row" role="row">
			<!-- Links start -->
			<article id="links">
				<h2>Links</h2>
				<div class="line"></div>
				<div class="articleBody">
					<p><a href="http://www.uidaho.edu/ibest">Institute for Bioinformatics and Evolutionary STudies (IBEST)</a></p>
					<p><a href="http://cores.ibest.uidaho.edu">IBEST Core Facilities</a></p>
					<p><a href="http://www.uidaho.edu/bcb">UI Bioinformatics and Computational Biology Graduate Program</a></p>
				</div>
			</article>
			<!-- links end -->
			<div class="clearfix"></div>
		</div>

		<!-- Row 5 -->
		<div id="row" role="row">
			<!-- publications start -->
			<article id="publications">
				<h2>Publications</h2> 
				<h3>
				<p>Total Publications: 47.</p>
				<p>h-index: 12, total citations: 456 on 40 publications via Web of Knowledge, <span id='badgeCont840364' style='width:26px'><script src='http://labs.researcherid.com/mashlets?el=badgeCont840364&mashlet=badge&showTitle=false&className=a&rid=F-2031-2011&size=small'></script></span>.</p>
				<p>h-index: 14, total citations: 625 on 34 articles via Google Scholar, <a href=http://scholar.google.com/citations?user=uojKM0oAAAAJ>Google Scholar</a>.</p>
				<p>Updated: November 6, 2013.</p>
				</h3>
				<div class="line"></div>
				<div class="articleBody">
					<h4>In Review</h4>
					<h4>2014</h4>
					<p>47. Yahvah KM, Brooker SL, Williams JE, <strong>Settles ML</strong>, McGuire MA, McGuire MK (2014) <dfn>Elevated dairy fat intake in lactating women alters milk lipid and fatty acids without detectible changes in expression of genes related to lipid uptake or synthesis</dfn>. Nutrition Research, Accepted Dec, 2014. In Press</p>
					<p>46. Liang S, Gliniewicz K, Mendes-Soares H, <strong>Settles ML</strong>, Forney L, Coats E, McDonald A (2014) <dfn>Comparative Analysis of Microbial Community of Novel Lactic Acid Fermentation Inoculated with Different Undefined Mixed Cultures.</dfn> Bioresource Technology, Accepted Dec, 2014. In Press.</p>
					<p>45. Dai J, Gliniewicz K, <strong>Settles ML</strong>, Coats ER, McDonald, AG (2014) <dfn>Influence of organic loading rate and solid retention time on polyhydroxybutyrate production from hybrid poplar hydrolysates using mixed microbial cultures.</dfn> Bioresource Technology, 175C:23-33. <a href=http://www.ncbi.nlm.nih.gov/pubmed/25459800>PMID: 25459800</a></p>
					<p>44. Li H, Ye D, Wang X, <strong>Settles ML</strong>, Wang J, Hao Z, Zhou L, Dong P, Jiang Y, Ma Z (2014) <dfn>Soil bacterial communities of different natural forest types in Northeast China.</dfn> Plant and Soil, 383(1-2):203-216.</p>
					<p>43. Dhimolea E, Wadia P, Murray T, <strong>Settles ML</strong>, Calafat AM, Treitman JD, Sonnenschein C, Shioda T, Soto A (2014) <dfn>Prenatal exposure to BPA alters the epigenome of the rat mammary gland and increases the propensity to neoplastic development.</dfn> PLoS One. 9(7):e99800. <a href=http://www.ncbi.nlm.nih.gov/pubmed/24988533>PMID: 24988533</a></p>
					<p>42. Jacobson JC, Turok DK, Dermish AI, Nygaard IE, <strong>Settles ML</strong>. (2014) <dfn>Vaginal microbiome changes with levonorgestrel intrauterine system placement.</dfn> Contraception, 90(2):130-135. <a href=http://www.ncbi.nlm.nih.gov/pubmed/24835828>PMID: 24835828</a></p>
					<p>41. Dziewanowska K, Settles M, Hunter S, Linquist I, Schilkey F, Hartzel PL. (2014) <dfn>Phase variation in Myxococcus xanthus yields cells specialized for iron sequestration.</dfn> PLoS One. 9(4):e95189. <a href=http://www.ncbi.nlm.nih.gov/pubmed/24733297>PMID: 24733297</a></p>
					<p>40. Hunter SS, Yano H, Loftie-Eaton W, Hughes J, De Gelder L, Stragier P, De Vos P, Set- tles ML, Top EM. (2014) <dfn>Draft genome sequence of Pseudomonas moraviensis R28-S.</dfn> Genome Announc. 2(1):e00035-14. <a href=http://www.ncbi.nlm.nih.gov/pubmed/24558233>PMID: 24558233</a></p>
					<p>39. Chapalamadugu KC, VandeVoort CA, <strong>Settles ML</strong>, Robison BD, Murdoch GK (2014) <dfn>Maternal Bisphenol A Exposure Impacts the Fetal Heart Transcriptome.</dfn> PLoS ONE 9(2): e89096. <a href=http://www.ncbi.nlm.nih.gov/pubmed/24586524>PMID: 24586524</a></p>

					<h4>2013</h4>
					<p>38. Hou, D, Zhou, X, Zhong, X, <strong>Settles, ML</strong>, Herring, J, Wang, L, Abdo, Z, Xu, C, Forney, LJ. <dfn>Microbiota of the seminal fluid from healthy and infertile men</dfn> Accepted to Journal of Clinical Microbiology. July 2013. <a href=http://www.ncbi.nlm.nih.gov/pubmed/23993888>PMID: 23993888</a></p> 
					<p>37. Zhbannikov I.Y.; Hunter S.S; <strong>Settles M.L.</strong>; Foster J.A. (2013) <dfn>SlopMap: A Software Applica- tion Tool for Quick and Flexible Identification of Similar Sequences Using Exact K-Mer Matching.</dfn> J Data Mining Genomics Proteomics 4:133.</p>
					<p>36. Benner, M, <strong>Settles, ML</strong>, Murdoch, G, Hardy, R, Robison, BD. (2013)<dfn>Sex specific transcriptional responses of the zebrafish (Danio rerio) brain selenoproteome to acute sodium selenite supplementation.</dfn> Physiological Genomics 45(15):653-666. <a href=http://www.ncbi.nlm.nih.gov/pubmed/23737534>PMID: 23737534</a></p>
					<p>35. Fremgen, S, Williams, A, Furusawa, G, Dziewanowska, K, <strong>Settles, M</strong>, Hartzell, P. <dfn>MasABK Proteins Interact with Proteins of the Type IV Pilin System to Affect Social Motility of Myxococcus xanthus.</dfn> PLoS ONE 8(1): e54557. (2013) <a href=http://www.ncbi.nlm.nih.gov/pubmed/23342171>PMID: 23342171</a></p>

					<h4>2012</h4>
					<p>34. Rosenblum EB, Poorten TJ, Joneson S, <strong>Settles M</strong>. <dfn>Substrate-Specific Gene Expression in Batrachochytrium dendrobatidis, the Chytrid Pathogen of Amphibians.</dfn> PLoS One, 7(11), e49924. (2012) <a href=http://www.ncbi.nlm.nih.gov/pubmed/23185485>PMID: 23185485</a></p>
					<p>33. Garrison, E, Treeck, M, Ehret, E, Butz, H, Garbuz, T, Oswald, BP, <strong>Settles, M</strong>, Boothroyd, J and Arrizabalaga, G. <dfn>A forward genetic screen reveals calcium-dependent protein kinase 3 regulates egress in Toxoplasma.</dfn> PLoS Pathogens. 8(11):e1003049. (2012). <a href=http://www.ncbi.nlm.nih.gov/pubmed/23209419>PMID: 23209419</a></p>
					<p>32. Guerrero-Bosagna, C., Covert, T., Haque, M., <strong>Settles, M.</strong>, Nilsson, E., Anway, M., Skinner, M. <dfn>Epigenetic Transgenerational Inheritance of Vinclozolin Induced Mouse Adult Onset Disease and Associated Sperm Epigenome Biomarkers.</dfn> Reproductive Toxicology, 34(4):694-707. (2012). <a href=http://www.ncbi.nlm.nih.gov/pubmed/23041264>PMID: 23041264</a></p>
					<p>31. <strong>Settles, ML</strong>, Coram, TE, Soule, T and Robison, BD <dfn>An improved algorithm for the detection of genomic variation using short oligonucleotide expression microarrays.</dfn> Molecular Ecology Resources, 12(6):1079â€“1089, (2012). <a href=http://www.ncbi.nlm.nih.gov/pubmed/22966828>PMID: 22966828</a></p>
					<p>30. Hughes, J, Lohman, B, Deckert, G, Nichols, E, <strong>Settles, M</strong>, Abdo, Z and Top, E <dfn>The Role of Clonal Interference in the Evolutionary Dynamics of Plasmid-Host Adaptation</dfn> mBio, 3(4):e00077-12. (2012). <a href=http://www.ncbi.nlm.nih.gov/pubmed/22761390>PMID: 22761390</a></p>
					<p>29. Drew, RE, <strong>Settles, ML</strong>, Churchill, EJ, Williams, SM, Balli, S and Robison, BD <dfn>Brain transcriptome variation among behaviorally distinct strains of zebrafish (Danio rerio).</dfn> BMC genomics, 13(1):323. (2012). <a href=http://www.ncbi.nlm.nih.gov/pubmed/22817472>PMID: 22817472</a></p>
					<p>28. Copeland, W, Krishnan, V, Beck, D, <strong>Settles, M</strong>, Foster, J, Cho, KC, Day, M, Hickey, R, Sch&uuml;tte, U, Zhou, X, Williams, C, Forney, L and Abdo, Z <dfn>mcaGUI: Microbial Community Analysis R-GUI.</dfn> Bioinformatics, 28(16):2198-2199. (2012). <a href=http://www.ncbi.nlm.nih.gov/pubmed/22692220>PMID: 22692220</a></p>
					<p>27. Minozzi, G, Williams, JL, Stella, A, Strozzi, F, Luini, MV, <strong>Settles, M</strong>, Taylor, JF, Whitlock, RH, Zanella, R and Neibergs, HL <dfn>Meta-analysis of Two Genome-Wide Association Studies of Bovine Paratuberculosis.</dfn> PLoS One, 7(3):e32578. (2012). <a href=http://www.ncbi.nlm.nih.gov/pubmed/22396781>PMID: 22396781</a></p>
					<p>26. Rosenblum, EB, Poorten, T, <strong>Settles, M</strong> and Murdoch, GK <dfn>Only skin deep: shared genetic response to the deadly chytrid fungus in susceptible frog species.</dfn> Molecular Ecology, 21(13):3110-3120. (2012). <a href=http://www.ncbi.nlm.nih.gov/pubmed/22332717>PMID: 22332717</a></p>
					<p>25. Turner, G, Cuthbertson, D, Voo, S, <strong>Settles, M</strong>, Grimes, H and Lange, M. <dfn>Experimental sink removal induces stress responses, including shifts in amino acid and phenylpropanoid metabolism, in soybean leaves.</dfn> Planta, 235(5):939-954. (2012). <a href=http://www.ncbi.nlm.nih.gov/pubmed/22109846>PMID: 22109846</a></p>

					<h4>2011</h4>
					<p>24. Murdoch, BM, Murdoch, GK, <strong>Settles, M</strong>, McKay, S, Williams, JL and Moore, S <dfn>Genome-Wide Scan Identifies Loci Associated with Classical BSE Occurrence.</dfn> PLoS ONE, 6(11):e26819. (2011). <a href=http://www.ncbi.nlm.nih.gov/pubmed/22073200>PMID: 22073200</a></p>
					<p>23. Furusawa, G, Dziewanowska, K, Stone, H, <strong>Settles, M</strong> and Hartzell, P. <dfn>Global analysis of phase variation in Myxococcus xanthus.</dfn> Molecular Microbiology, 81(3):784-804. (2011). <a href=http://www.ncbi.nlm.nih.gov/pubmed/21722202>PMID: 21722202</a></p>
					<p>22. Beck, D, <strong>Settles, M</strong> and Foster, J <dfn>OTUbase: an R infrastructure package for operational taxonomic unit data.</dfn> Bioinformatics, 27(12):1700-1701. (2011). <a href=http://www.ncbi.nlm.nih.gov/pubmed/21498398>PMID: 21498398</a>(*corresponding author)</p>
					<p>21. Zanella, R, <strong>Settles, M</strong>, McKay, S, Schnabel, R, Taylor, J,  Whitlock, R, Schukken, Y, Van Kessel, J, Smith, JM and Neibergs, H <dfn>Identification of Loci Associated with Tolerance to Johne's Disease in Holstein Cattle.</dfn> Animal Genetics, 42(1):28-38. (2011). <a href=http://www.ncbi.nlm.nih.gov/pubmed/20477805>PMID: 20477805</a></p>

					<h4>2010</h4>
					<p>20. Guerrero-Bosagna, C, <strong>Settles, M</strong>, Lucker, B and Skinner, MK. <dfn>Epigenetic Transgenerational Actions of Vinclozolin on Promoter Regions of the Sperm Epigenome.</dfn> PLoS ONE, 5(9):e13100. (2010). <a href=http://www.ncbi.nlm.nih.gov/pubmed/20927350>PMID: 20927350</a></p>
					<p>19. Neibergs, HL, <strong>Settles, ML</strong>, Whitlock, RH and Taylor, JF. <dfn>GSEA-SNP identifies genes associated with Johne's disease in cattle.</dfn> Mammalian genome, 21(7):419-425. (2010). <a href=http://www.ncbi.nlm.nih.gov/pubmed/20706723>PMID: 20706723</a></p>
					<p>18. Johnson, KA, Neibergs, H, Michal, JJ, Carstens, GE, <strong>Settles, M</strong>, Hafla, A, Forbes, TD, Holloway, JW, Brosh, A. <dfn>Differential expression of mitochondrial genes in liver from beef calves with divergent phenotypes for feed efficiency.</dfn> EAAP Scientific Series, 127(1):75-76. (2010).</p>
					<p>17. Clement TM, Savenkova MI, <strong>Settles, M</strong>, Anway MD and Skinner MK <dfn>Alterations in the developing testis   following embryonic vinclozolin exposure.</dfn> Reproductive Toxicology, 30(3):353-363. (2010). <a href=http://www.ncbi.nlm.nih.gov/pubmed/20566332>PMID: 20566332</a></p>
					<p>16. Murdoch, BM., Clawson, ML, Yue, S, Basu, U, McKay, S, <strong>Settles, M</strong>, Capoferri, R, Laegreid, WW, Williams, JL, Moore, SS <dfn>PRNP Haplotype Associated with Classical BSE Incidence in European Holstein Cattle.</dfn> PLoS ONE, 5(9):e12786. (2010). <a href=http://www.ncbi.nlm.nih.gov/pubmed/20862290>PMID: 20862290</a></p>
					<p>15. Murdoch, BM, Clawson, M, Laegreid, W, Stothard, P, <strong>Settles, M</strong>, McKay, S, Prasad, A, Wang, Z, Moore, SS and Williams, J <dfn>A 2cM genome-wide scan of European Holstein cattle affected by classical BSE.</dfn> BMC Genetics, 11(1). (2010). <a href=http://www.ncbi.nlm.nih.gov/pubmed/20350325>PMID: 20350325</a></p>
					<p>14. Coram, TE, Xueling, H, Zhan, G, <strong>Settles, ML</strong> and Chen, X <dfn>Meta-analysis of transcripts associated with race-specific resistance to stripe rust in wheat demonstrates common induction of blue copper-binding protein, heat-stress transcription factor, pathogen-induced WIR1A protein, and ent-kaurene synthase transcripts.</dfn> Functional and Integrative Genomics, 10(3):383-392. (2010). <a href=http://www.ncbi.nlm.nih.gov/pubmed/19937262>PMID: 19937262</a></p>

					<h4>2009</h4>
					<p>13. Rosenblum EB, Poorten TJ, <strong>Settles, M</strong>, Murdoch GK, Robert J, Maddox, N. and Eisen, MB. <dfn>Genome-Wide Transcriptional Response of Silurana (Xenopus) tropicalis to Infection with the Deadly Chytrid Fungus</dfn>. PLoS ONE 4(8):e6494. (2009). <a href=http://www.ncbi.nlm.nih.gov/pubmed/19701481>PMID: 19701481</a></p> 
					<p>12. Coram, TE, <strong>Settles, ML</strong> and Chen, X <dfn>Large-scale analysis of antisense transcription in wheat using the Affymetrix GeneChip Wheat Genome Array.</dfn> BMC Genomics, 10(1):253. (2009). <a href=http://www.ncbi.nlm.nih.gov/pubmed/19480707>PMID: 19480707</a></p>
					<p>11. <strong>Settles, ML</strong>, Zanella, R, McKay, SD, Schnabel, RD, Taylor, JF, Whitlock, R, Schukken, Y, Van Kessel, JS, Smith, JM and Neibergs, H <dfn>A whole genome association analysis identifies loci associated with Mycobacterium avium subsp. paratuberculosis infection status in US holstein cattle.</dfn> Animal Genetics, 40(5):655-662. (2009). <a href=http://www.ncbi.nlm.nih.gov/pubmed/19422364>PMID: 19422364</a></p>

					<h4>2008</h4>
					<p>10. Drew, RE, Rodnick, KJ, <strong>Settles, M</strong>, Wacyk, J, Churchill, E, Powell, M, Hardy, R, Murdoch, G, Hill, R and Robison, BD <dfn>Effect of Starvation on Gene Expression in Multiple Tissues of Zebrafish.</dfn> Physiological Genomics 35(3):283-295. (2008). <a href=http://www.ncbi.nlm.nih.gov/pubmed/18728227>PMID: 18728227</a></p>
					<p>9. Coram, TE, <strong>Settles, ML</strong>, Wang, M and Chen, X <dfn>Surveying expression level polymorphism and single-feature polymorphism in near-isogenic wheat lines differing for the Yr5  stripe rust resistance locus.</dfn> TAG Theoretical and Applied Genetics, 117(3):401-411. (2008). <a href=http://www.ncbi.nlm.nih.gov/pubmed/18470504>PMID: 18470504</a></p>
					<p>8. Robison BD, Drew, RE, Murdoch, G, Powell M, Rodnick KJ, <strong>Settles, M</strong>, Stone, D, Churchill, E, Hill, R, Papasani, M, Lewis, S and Hardy, R <dfn>Sexual dimorphism in hepatic gene expression and the response to dietary carbohydrate manipulation in the zebrafish (Danio rerio).</dfn> Comparative Biochemistry and Physiology Part D: Genomics and Proteomics, 3(2):141-154. (2008). <a href=http://www.ncbi.nlm.nih.gov/pubmed/20483215>PMID: 20483215</a></p>
					<p>7. Coram, TE, <strong>Settles, ML</strong> and Chen, X <dfn>Transcriptome analysis of high-temperature adult-plant resistance conditioned by Yr39 during the wheat-Puccinia striiformis f. sp. tritici interaction.</dfn> Molecular Plant Pathology, 9(4):479-493. (2008). <a href=http://www.ncbi.nlm.nih.gov/pubmed/18705862>PMID: 18705862</a></p>

					<h4>2005</h4>
					<p>6. <strong>Settles, M</strong> and Soule, T <dfn>Choosing an Algorithm: An Experimental Comparison of the Genetic Algorithm and Binary Particle Swarm.</dfn> In: Proceedings of the $17^{th}$ IMACS World Congress on Scientific Computation, Applied Mathematics and Simulation. Paris, France. (2005).</p>
					<p>5. <strong>Settles, M</strong> and Soule, T <dfn>Breeding Swarms: A GA/PSO Hybrid.</dfn> In: Proceeding of the Genetic and Evolutionary Computation Conference. pp. 161-168, Washington D.C. (2005).</p>
					<p>4. <strong>Settles, M</strong>, Nathan, P and Soule, T <dfn>Breeding Swarms: A New Approach to Recurrent Neural Network Training.</dfn> In: Proceeding of the Genetic and Evolutionary Computation Conference. pp. 185-192, Washington D.C. (2003).</p>

					<h4>2003</h4>
					<p>3. <strong>Settles, M</strong>, Rodebaugh, B and Soule, T <dfn>Comparison of genetic algorithm and particle swarm optimizer when evolving a recurrent neural network</dfn> In: Proceeding of the Genetic and Evolutionary Computation Conference. volume 2723, pp. 148-149, Chicago, IL. (2003).</p>
					<p>2. <strong>Settles, M</strong> and Soule, T <dfn>A Hybrid GA/PSO to Evolve Artificial Recurrent Neural Networks.</dfn> In: Intelligent Engineering Systems Through Artificial Neural Networks, volume 13, pp. 51-56, St. Louis, MO. (2003).</p>

					<h4>2002</h4>
					<p>1. <strong>Settles, M</strong> and Rylander, B <dfn>Neural network learning using particle swarm optimizers.</dfn> In: Advances in Information Science and Soft Computing, pp. 224-226. (2002). WSEAS Press.</p>
				</div>
			</article>
			<!-- publications end -->
			<div class="clearfix"></div>
		</div>

		<!-- Footer section -->
		<footer>
			<nav> <!-- The nav link semantically marks your main site navigation -->
				<ul>
					<li><a href="#contact">Contact</a></li>
					<li><a href="#research">Research/Interests</a></li>
					<li><a href="#courses">Courses</a></li>
					<li><a href="#software">Software</a></li>
					<li><a href="#links">Links</a></li>
					<li><a href="#publications">Publications</a></li>
				</ul>
			</nav>
			<div class="clearfix"></div>
			<div class="line"></div>
			<p>Copyright 2011 - Matt Settles</p> <!-- Change the copyright notice -->
			<a href="#" class="up">Go UP</a>
		</footer>
	</section> <!--! end of #page -->

	<!-- JavaScript at the bottom for fast page loading -->

	<!-- Grab Google CDN's jQuery, with a protocol relative URL; fall back to local if offline -->
	<script src="//ajax.googleapis.com/ajax/libs/jquery/1.6.2/jquery.min.js"></script>
	<script>window.jQuery || document.write('<script src="js/libs/jquery-1.6.2.min.js"><\/script>')</script>
	<!-- Smooth Scrolling with jquery.scollTo-min.js -->
	<script src="js/mylibs/jquery.scrollTo-1.4.2/jquery.scrollTo-min.js"></script>

	<!-- scripts concatenated and minified via ant build script-->
	<script defer src="js/plugins.js"></script>
	<script defer src="js/script.js"></script>
	<!-- end scripts-->

<!-- Change UA-XXXXX-X to be your site's ID -->
	<script>
		window._gaq = [['_setAccount','UA-28001006-1'],['_trackPageview'],['_trackPageLoadTime']];
		Modernizr.load({
			load: ('https:' == location.protocol ? '//ssl' : '//www') + '.google-analytics.com/ga.js'
		});
	</script>

	<!-- Prompt IE 6 users to install Chrome Frame. Remove this if you want to support IE 6.
		chromium.org/developers/how-tos/chrome-frame-getting-started -->
	<!--[if lt IE 7 ]>
	<script src="//ajax.googleapis.com/ajax/libs/chrome-frame/1.0.3/CFInstall.min.js"></script>
	<script>window.attachEvent('onload',function(){CFInstall.check({mode:'overlay'})})</script>
	<![endif]-->

</body>
</html>
