\documentclass[pdf]{beamer}
\usepackage[latin1]{inputenc}
\usepackage{multirow}
\usetheme{Antibes} %Warsaw
\usecolortheme{wolverine}

\usepackage{/Library/Frameworks/R.framework/Resources/share/texmf/tex/latex/Sweave}

\begin{document}

\title[Sequence Analysis]{Sequence Analysis in R}
\subtitle{BCB 504: Applied Bioinformatics\\}
\author[Matt Settles]{Matt Settles}
\institute{University of Idaho\\ Bioinformatics and Computational Biology Program}
\date{\today}


%% Title page
\begin{frame}[plain]
  \titlepage
\end{frame}


%% Outline
\begin{frame}[plain] 
  \frametitle{Outline}
  \tableofcontents
\end{frame}

\section{Introduction}
\begin{frame}
  \frametitle{What to use and when}
  Both R and Python have significant functionality to perform sequence analysis
  \begin{description}
  \item[Python] Biopython project - almost all biological sequence based
  \item[R] Bioconductor project (specifically Biostrings, BSgenome, GenomicFeatures, IRanges, ShortRead, Rsamtools, rSFFreader, and others)
  \end{description}

Both Python and R are usefull when you want to either preprocess data (ie cleanup, QA, etc.), or post processing of data (ie summarization, visualization and statistical analysis). When your dealing with one record, or read, at a time Python can be much more efficient and quicker. When you need analyize all data at once, or perform statistical assessment/analysis, R can be much more efficient.
\end{frame}

\section{IBEST tools}
\begin{frame}
  \frametitle{IBEST tools: modules, rSFFreader}
  need to first install git:  \href{http://git-scm.com/}{\textcolor{red}{http://git-scm.com/}}
  
  Then clone the package to your computer, from the command line
  \newline
  \alert{git clone git://github.com/msettles/rSFFreader.git}
  \newline
  \newline
  Install into R, from the command line
  \newline
  \alert{R CMD build rSFFreader}%
  \alert{R CMD INSTALL rSFFreadeR\_0.99.0.tar.gz}%
  \newline
  \newline
  The Genomics Resources Core module - grc
  \newline
  \href{http://help.ibest.uidaho.edu/index.php?pg=kb.book&id=2}{\textcolor{red}{IBEST computational help}}

\end{frame}

\section{Biostrings and BSgenome}
\begin{frame}[fragile]
\begin{footnotesize}
\begin{verbatim}
source("http://bioconductor.org/biocLite.R"); 
biocLite(c("Biostrings","BSgenome", "GenomicFeatures", 
"hgu95av2probe", "BSgenome.Celegans.UCSC.ce2", 
"BSgenome.Scerevisiae.UCSC.sacCer2", 
"BSgenome.Hsapiens.UCSC.hg19", 
"SNPlocs.Hsapiens.dbSNP.20101109",
"TxDb.Hsapiens.UCSC.hg19.knownGene", 
"TxDb.Scerevisiae.UCSC.sacCer2.ensGene"))
\end{verbatim}
\end{footnotesize}
\end{frame}

\begin{frame}
  \textbf{Biostrings}
  \begin{itemize}
  \item Data structures for representing large biological sequences (DNA/RNA/amino acids)
  \item Utilities for basic computations on sequences
  \item Tools for sequence matching and pairwise alignments
  \end{itemize}
  \textbf{BSgenome and genome packages}
  \begin{itemize}
  \item Full genomes stored in Biostring containers
  \item 16 organisms available via Bioconductor
  \item Facilities for supporting your own via BSgenomeForge
  \end{itemize}
\end{frame}

\begin{frame}
  \frametitle{Basic Sequence Classes}
  \begin{description}
  \item[Single sequence] BString, DNAString, RNAString, AAString
  \item[Set of sequences] BStringSet, DNAStringSet, RNAStringSet, AAStringSet, 
  \item[Views on sequences] Views
  \item[Masked sequences] MaskedBString, MaskedDNAString, MaskedRNAString, MaskedAAString
  \end{description}
\end{frame}

\begin{frame}[fragile]
 \frametitle{Constructing Sequences}
 \begin{block}{}
 \begin{footnotesize}
\begin{Schunk}
\begin{Sinput}
>  library(Biostrings)
>  dna <- DNAString("acctttGtGG-NNYaA")
>  dna
\end{Sinput}
\begin{Soutput}
  16-letter "DNAString" instance
seq: ACCTTTGTGG-NNYAA
\end{Soutput}
\begin{Sinput}
>  RNAString(dna)
\end{Sinput}
\begin{Soutput}
  16-letter "RNAString" instance
seq: ACCUUUGUGG-NNYAA
\end{Soutput}
\begin{Sinput}
>  DNA_ALPHABET
\end{Sinput}
\begin{Soutput}
 [1] "A" "C" "G" "T" "M" "R" "W" "S" "Y" "K" "V" "H" "D" "B"
[15] "N" "-" "+"
\end{Soutput}
\end{Schunk}
 \end{footnotesize}
 \end{block}
\end{frame}

\begin{frame}[fragile]
 \frametitle{Constructing Sets of Sequences}
 \begin{block}{}
 \begin{footnotesize}
\begin{Schunk}
\begin{Sinput}
>  library(Biostrings)
>  dnaset <- DNAStringSet(c("acctttGtGG-NNYaA","GATTACA"))
>  dnaset
\end{Sinput}
\begin{Soutput}
  A DNAStringSet instance of length 2
    width seq
[1]    16 ACCTTTGTGG-NNYAA
[2]     7 GATTACA
\end{Soutput}
\begin{Sinput}
>  names(dnaset) <- c("DNA1","DNA2")  
>  dnaset
\end{Sinput}
\begin{Soutput}
  A DNAStringSet instance of length 2
    width seq                           names               
[1]    16 ACCTTTGTGG-NNYAA              DNA1
[2]     7 GATTACA                       DNA2
\end{Soutput}
\end{Schunk}
 \end{footnotesize}
 \end{block}
\end{frame}

\begin{frame}[fragile]
  \frametitle{Basic Transformations}
  \begin{block}{}
  \begin{footnotesize}
\begin{Schunk}
\begin{Sinput}
> dna <- DNAString("TCAACGTTGAATAGCGTACCG")
> reverseComplement(dna)
\end{Sinput}
\begin{Soutput}
  21-letter "DNAString" instance
seq: CGGTACGCTATTCAACGTTGA
\end{Soutput}
\begin{Sinput}
> translate(dna)
\end{Sinput}
\begin{Soutput}
  7-letter "AAString" instance
seq: STLNSVP
\end{Soutput}
\begin{Sinput}
> translate(reverseComplement(dna))
\end{Sinput}
\begin{Soutput}
  7-letter "AAString" instance
seq: RYAIQR*
\end{Soutput}
\end{Schunk}
  \end{footnotesize}
  \end{block}
\end{frame}


\begin{frame}[fragile]
  \frametitle{Alphabets}
  \begin{block}{}
  \begin{footnotesize}
\begin{Schunk}
\begin{Sinput}
> alphabetFrequency(dna)
\end{Sinput}
\begin{Soutput}
A C G T M R W S Y K V H D B N - + 
6 5 5 5 0 0 0 0 0 0 0 0 0 0 0 0 0 
\end{Soutput}
\begin{Sinput}
> dinucleotideFrequency(dna)
\end{Sinput}
\begin{Soutput}
AA AC AG AT CA CC CG CT GA GC GG GT TA TC TG TT 
 2  2  1  1  1  1  3  0  1  1  0  2  2  1  1  1 
\end{Soutput}
\begin{Sinput}
> trinucleotideFrequency(dna)[1:30]
\end{Sinput}
\begin{Soutput}
AAA AAC AAG AAT ACA ACC ACG ACT AGA AGC AGG AGT ATA ATC ATG 
  0   1   0   1   0   1   1   0   0   1   0   0   1   0   0 
ATT CAA CAC CAG CAT CCA CCC CCG CCT CGA CGC CGG CGT CTA CTC 
  0   1   0   0   0   0   0   1   0   0   0   0   2   0   0 
\end{Soutput}
\end{Schunk}
additionally uniqueLetters(dna) and oligonucleotideFrequency(dna)

  \end{footnotesize}
  \end{block}
\end{frame}

\begin{frame}[fragile]
\textbf{BSgenome packages}
\begin{block}{}
\begin{footnotesize}
\begin{Schunk}
\begin{Sinput}
> library(BSgenome.Celegans.UCSC.ce2)
> #Celegans
> chrI <- Celegans$chrI
> chrI
\end{Sinput}
\begin{Soutput}
  15080483-letter "DNAString" instance
seq: GCCTAAGCCTAAGCCTAAGCCTAAGC...GCTTAGGCTTAGGTTTAGGCTTAGGC
\end{Soutput}
\end{Schunk}
\end{footnotesize}
\end{block}
\end{frame}


\begin{frame}[fragile]
\textbf{Input/Output of Sequences}
\begin{block}{}
\begin{footnotesize}
\begin{Schunk}
\begin{Sinput}
> seqs <- read.DNAStringSet(
+   system.file("extdata","fastaEx.fa",package="Biostrings"))
> seqs
\end{Sinput}
\begin{Soutput}
  A DNAStringSet instance of length 2
    width seq                           names               
[1]    72 AGTACGTAGTCGC...GACTAAACGATGC sequence1
[2]    70 AAACGATCGATCG...TACTGCATGCGGG sequence2
\end{Soutput}
\begin{Sinput}
> write.XStringSet(seqs,file="outseqs.fasta")
\end{Sinput}
\end{Schunk}
additionally read.BStringSet,read.AAStringSet,read.RNAStringSet\\
\textbf{Subsequences}
\begin{Schunk}
\begin{Sinput}
> subseq(seqs,start=c(5,9),end=c(10,35))
\end{Sinput}
\begin{Soutput}
  A DNAStringSet instance of length 2
    width seq                           names               
[1]     6 CGTAGT                        sequence1
[2]    27 GATCGTACTCGACTGATGTAGTATATA   sequence2
\end{Soutput}
\end{Schunk}
\end{footnotesize}
\end{block}
\end{frame}

\section{ShortRead}
\begin{frame}[fragile]
\textbf{ShortRead Package}
\begin{block}{}
\begin{footnotesize}
\begin{Schunk}
\begin{Sinput}
> library(ShortRead)
> fq <- readFastq(
+   system.file("unitTests/cases","sanger.fastq",package="ShortRead"))
> fq
\end{Sinput}
\begin{Soutput}
class: ShortReadQ
length: 1 reads; width: 27 cycles
\end{Soutput}
\end{Schunk}
\begin{Schunk}
\begin{Sinput}
> quality(fq)
> sread(fq)
> id(fq)
\end{Sinput}
\end{Schunk}
\end{footnotesize}
\end{block}
\end{frame}

\section{rSFFreader}
\begin{frame}
\begin{itemize}
\item SFF files are the raw data format for 'pyrosequencing' like data (ie Roche 454 and Ion Torrent)
\item Roche software provides for 2 tools: sffinfo sfffile
\item To use these tools you must load the module roche454
\item SFF files contain run information, read information, sequence, qualities and flowgrams
\end{itemize}
\end{frame}

\begin{frame}[fragile]
\textbf{rSFFreader Package}
\begin{block}{}
\begin{footnotesize}
\begin{Schunk}
\begin{Sinput}
> library(rSFFreader)
> sff <- readsff(
+  system.file("extdata","SmallTest.sff",package="rSFFreader"))
\end{Sinput}
\begin{Soutput}
Total number of reads to be read: 100
reading header for sff file:/Library/Frameworks/R.framework/Versions/2.14/Resources/library/rSFFreader/extdata/SmallTest.sff
reading file:/Library/Frameworks/R.framework/Versions/2.14/Resources/library/rSFFreader/extdata/SmallTest.sff
\end{Soutput}
\begin{Sinput}
> sff
\end{Sinput}
\begin{Soutput}
class: SffReadsQ
file: /Library/Frameworks/R.framework/Versions/2.14/Resources/library/rSFFreader/extdata/SmallTest.sff ; number of reads: 100 ; numer of flows: 800 
length: 100 reads; width: 51..428 basepair; clipping mode: Full 
\end{Soutput}
\begin{Sinput}
> clipMode(sff) <- "Raw"
> sff
\end{Sinput}
\begin{Soutput}
class: SffReadsQ
file: /Library/Frameworks/R.framework/Versions/2.14/Resources/library/rSFFreader/extdata/SmallTest.sff ; number of reads: 100 ; numer of flows: 800 
length: 100 reads; width: 80..458 basepair; clipping mode: Raw 
\end{Soutput}
\end{Schunk}
\end{footnotesize}
\end{block}
\end{frame}

\section{String Matching}
\begin{frame}
A common problem: find all the occurences (aka matches or hits) of a given pattern (typically short) in a (typically long) reference sequence (aka the subject) 
\begin{itemize}
  \item \textbf{matchPattern}: \textbf{1} pattern, \textbf{1} subject (reference sequence)
  \item \textbf{vmatchPattern}: \textbf{1} pattern, \textbf{N} subject (reference sequence)
  \item \textbf{matchPDict}: \textbf{N} pattern, \textbf{1} subject (reference sequence)
  \item \textbf{vmatchPDict}: \textbf{N} pattern, \textbf{N} subject (reference sequence)
\end{itemize}
\alert{pDict functions have 2 major limitations}
\begin{itemize}
  \item Dictionary must be preprocessed first
  \item must be constant width
\end{itemize}
\end{frame}

\begin{frame}[fragile]
\textbf{matchPattern}
\begin{block}{}
\begin{footnotesize}
\begin{Schunk}
\begin{Sinput}
> library(Biostrings)
> library(BSgenome.Celegans.UCSC.ce2)
> matchPattern("CAACTCCGATCG", Celegans$chrII)
\end{Sinput}
\begin{Soutput}
  Views on a 15279308-letter DNAString subject
subject: CCTAAGCCTAAGCCTAAGCCTAAG...CTTAGGCTTAGGCTTAGGCTTAGT
views:
       start      end width
[1] 13490043 13490054    12 [CAACTCCGATCG]
\end{Soutput}
\end{Schunk}
\end{footnotesize}
\end{block}
\end{frame}

\begin{frame}[fragile]
\textbf{matchPattern}
\begin{block}{}
\begin{footnotesize}
\begin{Schunk}
\begin{Sinput}
> matchPattern("CAACTCCGATCG", Celegans$chrII, max.mismatch=1)
\end{Sinput}
\begin{Soutput}
  Views on a 15279308-letter DNAString subject
subject: CCTAAGCCTAAGCCTAAGCCTAAG...CTTAGGCTTAGGCTTAGGCTTAGT
views:
        start      end width
 [1]   448786   448797    12 [CAAATCCGATCG]
 [2]  1258669  1258680    12 [CAACTCCGATGG]
 [3]  3340998  3341009    12 [CAGCTCCGATCG]
 [4]  3441302  3441313    12 [CACCTCCGATCG]
 [5]  4059036  4059047    12 [CAACTCCGATCT]
 [6]  4588202  4588213    12 [CAACTTCGATCG]
 [7]  7209941  7209952    12 [CAACTCCGATCC]
 [8]  9946308  9946319    12 [CAACTCCGATCC]
 [9] 11068482 11068493    12 [CAACTCCGATTG]
[10] 11304102 11304113    12 [CAACTCCGATCT]
[11] 13490043 13490054    12 [CAACTCCGATCG]
[12] 13760610 13760621    12 [CAACTCCGATTG]
[13] 15213851 15213862    12 [CAACTCCGATCT]
\end{Soutput}
\end{Schunk}
\end{footnotesize}
\end{block}
\end{frame}

\begin{frame}[fragile]
\textbf{matchPattern}
\begin{block}{}
\begin{footnotesize}
\begin{Schunk}
\begin{Sinput}
> matchPattern("CAACTCCGATCG", Celegans$chrII, max.mismatch=1, with.indels=TRUE)
\end{Sinput}
\begin{Soutput}
  Views on a 15279308-letter DNAString subject
subject: CCTAAGCCTAAGCCTAAGCCTAAG...CTTAGGCTTAGGCTTAGGCTTAGT
views:
        start      end width
 [1]   448786   448797    12 [CAAATCCGATCG]
 [2]   861918   861928    11 [CAACTCCGATG]
 [3]  1258669  1258679    11 [CAACTCCGATG]
 [4]  1947047  1947057    11 [CAACTCCATCG]
 [5]  2022293  2022303    11 [CAACTCCATCG]
 [6]  2517033  2517043    11 [CAACTCCATCG]
 [7]  2658084  2658094    11 [CACTCCGATCG]
 [8]  2839525  2839535    11 [CAACTCCGATG]
 [9]  3340998  3341009    12 [CAGCTCCGATCG]
 ...      ...      ...   ... ...
[30] 10984646 10984658    13 [CAACTCCGATTCG]
[31] 11068482 11068493    12 [CAACTCCGATTG]
[32] 11304102 11304112    11 [CAACTCCGATC]
[33] 11576241 11576251    11 [CACTCCGATCG]
[34] 12895956 12895966    11 [CAACTCCGACG]
[35] 13383417 13383427    11 [CAACTCCATCG]
[36] 13490043 13490054    12 [CAACTCCGATCG]
[37] 13760610 13760621    12 [CAACTCCGATTG]
[38] 15213851 15213861    11 [CAACTCCGATC]
\end{Soutput}
\end{Schunk}
\end{footnotesize}
\end{block}
\end{frame}

\begin{frame}[fragile]
\textbf{matchPDict}
\begin{block}{}
\begin{footnotesize}
\textbf{load the dictionary}
\begin{Schunk}
\begin{Sinput}
> library(hgu95av2probe)
> dict0 <- DNAStringSet(hgu95av2probe)
> dict0
\end{Sinput}
\begin{Soutput}
  A DNAStringSet instance of length 201800
         width seq
     [1]    25 TGGCTCCTGCTGAGGTCCCCTTTCC
     [2]    25 GGCTGTGAATTCCTGTACATATTTC
     [3]    25 GCTTCAATTCCATTATGTTTTAATG
     [4]    25 GCCGTTTGACAGAGCATGCTCTGCG
     [5]    25 TGACAGAGCATGCTCTGCGTTGTTG
     [6]    25 CTCTGCGTTGTTGGTTTCACCAGCT
     [7]    25 GGTTTCACCAGCTTCTGCCCTCACA
     [8]    25 TTCTGCCCTCACATGCACAGGGATT
     [9]    25 CCTCACATGCACAGGGATTTAACAA
     ...   ... ...
[201792]    25 GAGTGCCAATTCGATGATGAGTCAG
[201793]    25 ACACTGACACTTGTGCTCCTTGTCA
[201794]    25 CAATTCGATGATGAGTCAGCAACTG
[201795]    25 GACTTTCTGAGGAGATGGATAGCCT
[201796]    25 AGATGGATAGCCTTCTGTCAAAGCA
[201797]    25 ATAGCCTTCTGTCAAAGCATCATCT
[201798]    25 TTCTGTCAAAGCATCATCTCAACAA
[201799]    25 CAAAGCATCATCTCAACAAGCCCTC
[201800]    25 GTGCTCCTTGTCAACAGCGCACCCA
\end{Soutput}
\end{Schunk}
\end{footnotesize}
\end{block}
\end{frame}

\begin{frame}[fragile]
\textbf{matchPDict}
\begin{block}{}
\begin{footnotesize}
\textbf{preprocess the dictionary}
\begin{Schunk}
\begin{Sinput}
> pdict <- PDict(dict0)
> pdict
\end{Sinput}
\begin{Soutput}
TB_PDict object of length 201800 and width 25 (preprocessing algo="ACtree2")
\end{Soutput}
\end{Schunk}
\textbf{load the subject}
\begin{Schunk}
\begin{Sinput}
> library(BSgenome.Hsapiens.UCSC.hg19)
> chr1 <- unmasked(Hsapiens$chr1)
> chr1
\end{Sinput}
\begin{Soutput}
  249250621-letter "DNAString" instance
seq: NNNNNNNNNNNNNNNNNNNNNNNNNN...NNNNNNNNNNNNNNNNNNNNNNNNNN
\end{Soutput}
\end{Schunk}
\textbf{call matchPDict}
\begin{Schunk}
\begin{Sinput}
> m <- matchPDict(pdict, chr1)
> m
\end{Sinput}
\begin{Soutput}
MIndex object of length 201800
\end{Soutput}
\end{Schunk}
\end{footnotesize}
\end{block}
\end{frame}


\begin{frame}[fragile]
\textbf{matchPDict}
\begin{block}{}
\begin{footnotesize}
\textbf{query the object}
\begin{Schunk}
\begin{Sinput}
> m[[700]] # results for the 700th pattern
\end{Sinput}
\begin{Soutput}
IRanges of length 3
        start       end width
[1]  59015037  59015061    25
[2] 110955918 110955942    25
[3] 197066271 197066295    25
\end{Soutput}
\begin{Sinput}
> startIndex(m)[[700]]
\end{Sinput}
\begin{Soutput}
[1]  59015037 110955918 197066271
\end{Soutput}
\begin{Sinput}
> endIndex(m)[[700]] 
\end{Sinput}
\begin{Soutput}
[1]  59015061 110955942 197066295
\end{Soutput}
\end{Schunk}
\end{footnotesize}
\end{block}
\end{frame}

\begin{frame}
More string matching functions\\
countPattern, vCountPattern, countPDict, vcountPDict: counts the number of matches
\begin{small}
\begin{itemize}
  \item \textbf{trimLRPatterns}: trim left and/or right patterns from sequences
  \item \textbf{matchLRPatterns}: the matches are specified by a left pattern, a right pattern and a maximum distance between them
  \item \textbf{matchProbePair}: finds amplicons given by a pair of primers (simulate PCR)
  \item \textbf{matchPWM}: finds motifs described by a Position Weight Matrix (PWM)
  \item \textbf{findPalindromes/findComplimentedPalindromes}
  \item \textbf{pairwiseAlignment}: solves (NeedlemanWunsch) global alignment, (Smith Waterman) local alignement, and (endsfree) overlap alignment problems
\end{itemize}
\end{small}
not all of these support with.indels \textbf{yet}, see manuals for what is supported at this time 
\end{frame}

\begin{frame}[fragile]
\textbf{using trimLRPattern}
\begin{block}{}
\begin{footnotesize}
\begin{Schunk}
\begin{Sinput}
> subject <- DNAStringSet(c("TGCTTGACGCAAAGA", "TTCTGCTTGGATCGG"))
> subject
\end{Sinput}
\begin{Soutput}
  A DNAStringSet instance of length 2
    width seq
[1]    15 TGCTTGACGCAAAGA
[2]    15 TTCTGCTTGGATCGG
\end{Soutput}
\begin{Sinput}
> trimLRPatterns(Lpattern="TTCTGCTT", Rpattern="ATCGGAAG", subject)
\end{Sinput}
\begin{Soutput}
  A DNAStringSet instance of length 2
    width seq
[1]     9 GACGCAAAG
[2]     2 GG
\end{Soutput}
\end{Schunk}
\end{footnotesize}
\end{block}
\end{frame}


\begin{frame}[fragile]
\textbf{using pairwiseAlignment}
\begin{block}{}
\begin{footnotesize}
\begin{Schunk}
\begin{Sinput}
> pairwiseAlignment("TTGCACCC", "TTGGATTGACCCA")
\end{Sinput}
\begin{Soutput}
Global PairwiseAlignedFixedSubject (1 of 1)
pattern: [1] TTGCA-----CCC 
subject: [1] TTGGATTGACCCA 
score: -29.90804 
\end{Soutput}
\begin{Sinput}
> pairwiseAlignment("TTGCACCC", "TTGGATTGACCCA", type="global-local")
\end{Sinput}
\begin{Soutput}
Global-Local PairwiseAlignedFixedSubject (1 of 1)
pattern: [1] TTGCACCC 
subject: [6] TTG-ACCC 
score: -0.1277071 
\end{Soutput}
\begin{Sinput}
> pairwiseAlignment("TTC", "ATTATTA", type="global-local")
\end{Sinput}
\begin{Soutput}
Global-Local PairwiseAlignedFixedSubject (1 of 1)
pattern: [1] TTC 
subject: [5] TTA 
score: -2.596666 
\end{Soutput}
\end{Schunk}
\end{footnotesize}
\end{block}
\end{frame}
\end{document}
